joomla/joomla-cms
# [5.2] Implement Schema.org support for Menu Items

### Summary of Changes
Implements same Article and Contact Schema.org parameters and parsing into JSON-LD for Menu Items. This allows extensions and other custom pages to leverage Schema.org JSON-LD without needing any additional development.

The JSON-LD will be appended in the following order: Base -> Webpage -> Menu -> Content

Menu Schema can be globally turned off should anyone not want it. Default enabled, but same as articles it requires the schemaorg plugin to be saved at least once to establish the base.


### Testing Instructions
1. Edit, publish, toggle on Menus parameter, and save "System - Schema.org" plugin
2. Create or edit a new menu item
3. Configure and save data under the Schema tab as you would with an Article
4. Navigate to that menu item on frontend
5. Press F12 to open your browsers developer tools
6. Review the `<head>` to make sure your JSON-LD Schema data is present

I've tested this on the following component frontend views: com_content, com_users, com_contact, and com_wrapper

Additional component tests would be appreciated (e.g. a variant of 3rd party extensions would be great).


### Actual result BEFORE applying this Pull Request
No Schema tab in menu edit/create


### Expected result AFTER applying this Pull Request
Schema tab in menu edit/create


### Link to documentations
Please select:
